marconi/test/Spec/Utxo.hs

```diff
@@ -54,31 +170,40 @@ txToUtxoTest = property $ do
         Just targets    ->
             ( nub
               . catMaybes
-              . fmap (\x -> addressAnyToShelley (x ^. Utxo.utxoAddress))
+              . fmap (\x -> storableQueryToShelleyAddr (x ^. Utxo.address))
               $ utxos) === (nub . toList $ targets)
 
-genBlockNo :: Gen C.BlockNo
-genBlockNo = C.BlockNo <$> Gen.word64 Range.constantBounded
-
-genEvents :: Gen Utxo.UtxoEvent
-genEvents = do
-    slotNo <- CGen.genSlotNo
-    blockNo  <- genBlockNo
-    txs <- Gen.list (Range.linear 2 5)(CGen.genTx C.ShelleyEra)
-    pure . fromJust $ Utxo.getUtxoEvents Nothing slotNo blockNo txs
-
 utxoStorageTest :: Property
 utxoStorageTest = property $ do
-    depth <- forAll $ Gen.int (Range.linear 1  5) -- force DB writes
-    ndx <- liftIO $ Utxo.open ":memory:" (Utxo.Depth depth)
-    events <- forAll $ Gen.list (Range.linear 1 10) genEvents
-    let (rows :: [Utxo.UtxoRow]) = nub . concatMap Utxo.toRows $ events
-        (addresses :: [C.AddressAny]) = nub . fmap (\r -> r ^. Utxo.utxoRowUtxo . Utxo.utxoAddress ) $ rows
-    ix <- liftIO $ Ix.insertL (events) ndx
-    rowsFromStore <- liftIO $ (concat . catMaybes) <$> forM addresses (\a -> Utxo.queryPlusVolatile ix a  )
-    (null rowsFromStore) === False
-    let (tid :: Utxo.UtxoRow -> C.AddressAny) = (\r -> r ^. Utxo.utxoRowUtxo . Utxo.utxoAddress)
-    (all (`elem` (tid <$> rows)) (tid <$> rowsFromStore)) === True
+    depth <- forAll $ Gen.int (Range.linear 2  3) -- force DB writes
+    indexer <- liftIO $ Utxo.open ":memory:" (Utxo.Depth depth)
+    events <- forAll $ Gen.list (Range.linear 2 5 )genEvents
+    let (eUtxos :: [Utxo.Utxo]) = concatMap (Set.toList . Utxo.ueUtxos) events
+    let qs :: [StorableQuery Utxo.UtxoHandle]
+        qs = nub $ eUtxos & toListOf (folded . Utxo.address)
+    (void . liftIO . Storable.insertMany events)  indexer
+    (fromStorage :: [StorableEvent Utxo.UtxoHandle]) <- (liftIO . Storable.getEvents) indexer
+      -- liftIO . traverse (Storable.query Storable.QEverything indexer) $ qs
+    (fromMemory :: [StorableEvent Utxo.UtxoHandle]) <- liftIO $ (Storable.getMemoryEvents (indexer ^. Storable.storage )) & V.freeze <&> V.toList
+    -- let (rows :: [Utxo.UtxoRow]) =
+    --       concatMap (\Utxo.UtxoResult  rs -> rs) results
+    --     ces = Utxo.rowsToEvents rows
+    --     computedEvents =
+
+  -- TODO bad tests, remove once we resolve the bug
```

Comment:
@raduom this tests fails. The failure may be to my misunderstanding of the Rewindable API.
 It seems as if the getMemoryEvents may have a bug, or I am not using it correctly.  Here are my observation:
- for small number of events, the Storable.getEvents  returns no events.
- When I increase the number of events per test, the Storable.getEvents returns some events, less than expected
- The bug seem to be with Storable.getMemoryEvents 
- for sufficiently large number of events, events are flushed to DB
Strangely, I can see events being written to DB when I increase the number of events per test, however, the getEvents doesnot report
- The print statements are there to drive the point in the discrepancies.  I'll remove them once the issue is resolve.